src/main/java/dev/jbang/util/JavaUtil.java

```diff
@@ -123,13 +123,15 @@ public static String parseJavaOutput(String output) {
 		return null;
 	}
 
+	// Match the first number in a Java version string, except if
+	// it starts with "1.", in that case, match the second number
+	private static final String JVERSION = "^(?:1\\.)?(\\d+).*";
```

Comment:
This last commit is not really necessary IMO. The final static constant I introduced in my code was so the Pattern only needed to be compiled once. When using a String this advantage does not exist.